src/classes/RD_AddDonationsBTN_CTRL.cls

```diff
@@ -65,13 +53,13 @@ public with sharing class RD_AddDonationsBTN_CTRL {
     public PageReference buttonClick(){
         Savepoint sp = Database.setSavepoint();
         try {
-            system.assert(RecurringDonation != null);        
+            system.assert(sc.getId() != null);
 	        string qString = RD_RecurringDonations.strQueryRDNoWhere();
-            qString += ' where id = \'' + RecurringDonation.id + '\'';
-	        RecurringDonation = (npe03__Recurring_Donation__c)database.query(qString);
+            qString += ' where id = \'' + sc.getId() + '\'';
+	        npe03__Recurring_Donation__c RecurringDonation = (npe03__Recurring_Donation__c)database.query(qString);
 	        RD_RecurringDonations.evaluateRecurringDonationsForNewOppInsert(new list<npe03__Recurring_Donation__c>{RecurringDonation});
 	        
-	        PageReference pageRef = new PageReference(returnurl);
+	        PageReference pageRef = sc.view();
```

Comment:
@dospromptman Same issue with sc.view(): the page is unrefreshed most of the time. Wondering if it's worth including the t parameter hack at all, I imagine they'll have to fix this sooner rather than later.